lib/matplotlib/widgets.py

```diff
@@ -2966,100 +3056,129 @@ def _release(self, event):
     def _onmove(self, event):
         """Motion notify event handler."""
 
-        state = self._state | self._default_state
+        state = self._state
+        rotate = ('rotate' in state and
+                  self._active_handle in self._corner_order)
+        eventpress = self._eventpress
+        # The calculations are done for rotation at zero: we apply inverse
+        # transformation to events except when we rotate and move
+        move = self._active_handle == 'C'
+        if not move and not rotate:
+            inv_tr = self._get_rotation_transform().inverted()
+            event.xdata, event.ydata = inv_tr.transform(
+                [event.xdata, event.ydata])
+            eventpress.xdata, eventpress.ydata = inv_tr.transform(
+                [eventpress.xdata, eventpress.ydata]
+                )
+
+        dx = event.xdata - eventpress.xdata
+        dy = event.ydata - eventpress.ydata
+        # refmax is used when moving the corner handle with the square state
+        # and is the maximum between refx and refy
+        refmax = None
+        if self._use_data_coordinates:
+            refx, refy = dx, dy
+        else:
+            # Add 1e-6 to avoid divided by zero error
+            refx = event.xdata / (eventpress.xdata or eventpress.xdata + 1E-6)
+            refy = event.ydata / (eventpress.ydata or eventpress.ydata + 1E-6)
```

Comment:
```suggestion
            refx = event.xdata / (eventpress.xdata or 1E-6)
            refy = event.ydata / (eventpress.ydata or 1E-6)
```
We don't have to add 0 :).

Reply:
Indeed... :D